File: app/src/main/java/com/ragdroid/dahaka/activity/login/LoginComponent.java
package com.ragdroid.dahaka.activity.login;

import com.ragdroid.dahaka.activity.ActivityScope;

import dagger.BindsInstance;
import dagger.Subcomponent;

/**
 * Created by garimajain on 13/08/17.
 */
@ActivityScope
@Subcomponent(modules = LoginModule.class)
public interface LoginComponent {


    LoginActivity inject(LoginActivity loginActivity);

    @Subcomponent.Builder
    interface Builder {
        LoginComponent build();

        @BindsInstance
        Builder loginActivity(LoginActivity loginActivity);
    }

}

File: app/src/main/java/com/ragdroid/dahaka/activity/login/LoginComponent.kt
package com.ragdroid.dahaka.activity.login

import com.ragdroid.dahaka.activity.ActivityScope

import dagger.BindsInstance
import dagger.Subcomponent

/**
 * Created by garimajain on 13/08/17.
 */
@ActivityScope
@Subcomponent(modules = arrayOf(LoginModule::class))
interface LoginComponent {


    fun inject(loginActivity: LoginActivity): LoginActivity

    @Subcomponent.Builder
    interface Builder {
        fun build(): LoginComponent

        @BindsInstance
        fun loginActivity(loginActivity: LoginActivity): Builder
    }

}
